examples/mnist/convert_mnist_data.cpp

```diff
@@ -6,12 +6,22 @@
 // The MNIST dataset could be downloaded at
 //    http://yann.lecun.com/exdb/mnist/
 
+#ifdef USE_GFLAGS
 #include <gflags/gflags.h>
-#include <glog/logging.h>
+#endif   // USE_GFLAGS
+#ifdef USE_GLOG
+#include "glog/logging.h"
+#else
+#include "caffe/util/glog_alternate.hpp"
+#endif  // USE_GLOG
```

Comment:
Macros are ugly. 

Most headers could be accumulated in `caffe/dependencies.h` header. This solves this.

Most but not all, since should not force library users to include all dependencies for case if he doesn't need some of them.